src/exec/util/file.go

```diff
@@ -15,20 +15,119 @@
 package util
 
 import (
+	"bytes"
+	"compress/gzip"
+	"errors"
 	"io/ioutil"
+	"net/http"
 	"os"
 	"path/filepath"
 
 	"github.com/coreos/ignition/config/types"
+	"github.com/coreos/ignition/src/log"
+	"github.com/coreos/ignition/src/util"
+
+	"github.com/coreos/ignition/third_party/github.com/vincent-petithory/dataurl"
 )
 
 const (
-	DefaultDirectoryPermissions types.FileMode = 0755
-	DefaultFilePermissions      types.FileMode = 0644
+	DefaultDirectoryPermissions os.FileMode = 0755
+	DefaultFilePermissions      os.FileMode = 0644
+)
+
+var (
+	ErrSchemeUnsupported = errors.New("unsupported source scheme")
+	ErrStatusBad         = errors.New("bad HTTP response status")
 )
 
+type File struct {
+	Path     types.Path
+	Contents []byte
+	Mode     os.FileMode
+	Uid      int
+	Gid      int
+}
+
+func RenderFile(l *log.Logger, f types.File) *File {
+	var contents []byte
+
+	if l.LogOp(func() error {
+		switch f.Contents.Source.Scheme {
+		case "http":
+			client := util.NewHttpClient(l)
+			data, status, err := client.Get(f.Contents.Source.String())
+			if err != nil {
+				return err
+			}
+
+			l.Debug("GET result: %s", http.StatusText(status))
+			if status != http.StatusOK {
+				return ErrStatusBad
+			}
+
+			contents = data
+		case "data":
+			url, err := dataurl.DecodeString(f.Contents.Source.String())
```

Comment:
It's needed if someone wants to include the file contents inline in the config, rather than fetching the contents.